Cabal/Distribution/Parsec/Class.hs

```diff
@@ -0,0 +1,254 @@
+{-# LANGUAGE FlexibleContexts #-}
+module Distribution.Parsec.Class (
+    Parsec(..),
+    -- * Type aliases
+    ParsecBS,
+    ParsecString,
+    -- * Utilities
+    parsecToken,
+    parsecFilePath,
+    parsecQuoted,
+    parsecMaybeQuoted,
+    parsecCommaList,
+    parsecOptCommaList,
+    ) where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Distribution.Parsec.Types.Common (PWarning (..))
+import Data.Functor.Identity (Identity)
+
+import qualified Data.ByteString as BS
+import qualified Text.Parsec     as P
+import qualified Text.Parsec.Pos as P
+
+-- Instances
+import Distribution.Version (Version (..), VersionRange (..), noVersion, orEarlierVersion, orLaterVersion, anyVersion, unionVersionRanges, intersectVersionRanges, thisVersion, laterVersion, earlierVersion)
+import Distribution.Package (PackageName (..), Dependency (..))
+import Distribution.License (License (..))
+import Distribution.Types.BuildType (BuildType (..))
+import Distribution.ModuleName (ModuleName (..))
+
+import Distribution.Text (display)
+
+-------------------------------------------------------------------------------
+-- Class
+-------------------------------------------------------------------------------
+
+-- |
+--
+-- TODO: implementation details: should be careful about consuming trailing whitespace?
+-- Should we always consume it?
```

Comment:
For now `Parsec` class (is naming good btw?) is used only to parse fields, but it could (will?) be used for command line args as well, if we get rid of `ReadP` completely